miniapps/meshing/hpref_serial.cpp

```diff
@@ -0,0 +1,410 @@
+//                       Serial hp-refinement example
+//
+// Compile with: make hpref_serial
+//
+// Sample runs:  hpref_serial -dim 2 -n 1000
+//               hpref_serial -dim 3 -n 500
+//
+// Description:  This example demonstrates h- and p-refinement in a serial
+//               finite element discretization of the Laplace problem (cf. ex1)
+//               -Delta u = 1 with homogeneous Dirichlet boundary conditions.
+//               Refinements are performed iteratively, each iteration having h-
+//               or p-refinements. For simplicity, we randomly choose the
+//               elements and the type of refinement, for each iteration. In
+//               practice, these choices may be made in a problem-dependent way,
+//               but this example serves only to illustrate the capabilities of
+//               hp-refinement.
+
+//               We recommend viewing Example 1 before viewing this example.
+
+#include "mfem.hpp"
+#include <fstream>
+#include <iostream>
+
+using namespace std;
+using namespace mfem;
+
+real_t CheckH1Continuity(GridFunction & x);
+
+// Deterministic function for "random" integers.
+int DetRand(int & seed)
+{
+   seed++;
+   return int(std::abs(1.0e5 * sin(seed * 1.1234 * M_PI)));
+}
+
+void f_exact(const Vector &x, Vector &f);
+
+int main(int argc, char *argv[])
+{
+   // 1. Parse command-line options.
+   int order = 1;
+   bool static_cond = false;
+   bool pa = false;
+   bool fa = false;
+   const char *device_config = "cpu";
+   bool visualization = true;
+   bool algebraic_ceed = false;
+   int numIter = 0;
+   int dim = 2;
+   bool deterministic = true;
+   bool projectSolution = false;
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&order, "-o", "--order",
+                  "Finite element order (polynomial degree) or -1 for"
+                  " isoparametric space.");
+   args.AddOption(&static_cond, "-sc", "--static-condensation", "-no-sc",
+                  "--no-static-condensation", "Enable static condensation.");
+   args.AddOption(&pa, "-pa", "--partial-assembly", "-no-pa",
+                  "--no-partial-assembly", "Enable Partial Assembly.");
+   args.AddOption(&fa, "-fa", "--full-assembly", "-no-fa",
```

Comment:
It seems that `-pa` and `-fa` fail because of solver issues.  Also `-sc` fails possibly related to issues with the trace space. Maybe we can just remove these options.

Reply:
I removed these options. They were copied and pasted, but they are not supported.